resources/assets/js/components/Progress.js

```diff
@@ -0,0 +1,51 @@
+import React, {Component} from 'react';
+import axios from 'axios';
+import ReactDOM from 'react-dom';
+
+export default class Progress extends Component {
+  constructor() {
+    super();
+    this.state = {
+      subjects: []
+    };
+    this.loadProgress();
+  }
+
+  loadProgress() {
+    axios.get('/submissions/progress').then((response) => {
+      this.setState(response.data);
+    });
+  }
+
+  getRating(number) {
+    if (number >= 90) return "excellent";
+    if (number >= 70) return "good";
+    if (number >= 50) return "ok";
+    return "improvement";
```

Comment:
instead of improvement, do "needs improvement", as improvement sounds like they are actually making progress